lib/main.js

```diff
@@ -107,52 +131,84 @@ HTMLUglify.prototype.setNewSelector = function(lead, rule, toReplace, pointer) {
+          value = selectorPart.replace(/\#/g, '');
+
+          if (value && !self.isWhitelisted('id', value)) {
+            pointer = self.determinePointer('id', value, lookups);
+            self.populateLookups('id', value, pointer, lookups);
+          }
         }
 
         // handle .something
         if (selectorPart.indexOf('.') > -1) {
-          toReplace = selectorPart.replace(/\./g, '');
-          keyLookup = ['class', toReplace].join(LOOKUP_DELIMITER);
+          value = selectorPart.replace(/\./g, '');
+
+          if (value && !self.isWhitelisted('class', value)) {
+            pointer = self.determinePointer('class', value, lookups);
+            self.populateLookups('class', value, pointer, lookups);
+          }
         }
 
-        // handle *[id=something]
-        if (selectorPart.indexOf('id=') > -1){
-          toReplace = selectorPart.replace(/id=/g, '').replace(/\"/g, ''); // remove quotes from "somevalue"
-          keyLookup = ['id', toReplace].join(LOOKUP_DELIMITER);
+        // handle *[id=something] or *[id*=something]
+        if (
+          selectorPart.indexOf('id=') > -1 || 
+          selectorPart.indexOf('id*=') > -1 ||
+          selectorPart.indexOf('id^=') > -1 ||
+          selectorPart.indexOf('id$=') > -1
+        ) {
+          value = selectorPart.replace(/id\*?\^?\$?=|\"|\'/g, '');
+          if (value && !self.isWhitelisted('id', value)) {
+            pointer = self.determinePointer('id', value, lookups);
+            self.populateLookups('id', value, pointer, lookups);
+          }
         }
 
-        // handle *[class=something]
-        if (selectorPart.indexOf('class=') > -1) {
-          toReplace = selectorPart.replace(/class=/g, '');
-          keyLookup = ['class', toReplace].join(LOOKUP_DELIMITER);
+        // handle *[class=something] or *[class*=something]
+        if (
+          selectorPart.indexOf('class=') > -1 || 
+          selectorPart.indexOf('class*=') > -1 ||
+          selectorPart.indexOf('class^=') > -1 ||
+          selectorPart.indexOf('class$=') > -1
+        ) {
+          value = selectorPart.replace(/class\*?\^?\$?=|\"|\'/g, '');
+
+          if (value && !self.isWhitelisted('class', value)) {
+            pointer = self.determinePointer('class', value, lookups);
+            self.populateLookups('class', value, pointer, lookups);
+          }
         }
 
-        // handle *[for=something]
-        if (selectorPart.indexOf('for=') > -1){
-          toReplace = selectorPart.replace(/for=/g, '');
-          toReplace = toReplace.replace(/\"/g, ''); // remove quotes from "somevalue"
-          keyLookup = ['id', toReplace].join(LOOKUP_DELIMITER);
+        // handle *[for=something] or *[for*=something]
+        if (
+          selectorPart.indexOf('for=') > -1 || 
+          selectorPart.indexOf('for*=') > -1 ||
+          selectorPart.indexOf('for^=') > -1 ||
+          selectorPart.indexOf('for$=') > -1
```

Comment:
Indeed. 

My pull request here will actually be to ignore those ones right now.

+ `[attr~=value]` is a whitespace-separated list of words (which I haven't seen us need use for and looks tricky to implement)
+ `attr|=value` is very similar to to `*` so I just opted to leave it out.
+ `attr operator value i` Adding an i before the closing bracket causes the value to be compared case-insensitively. Opting to leave this out because we're not currently matching on case with uglify.

What's your thoughts?